docs/common/DocsPageTemplate/SideNav/index.vue

```diff
@@ -77,22 +79,54 @@
         return toc;
       },
     },
+
     watch: {
       filterText(newValue) {
         if (window) {
-          window.sessionStorage.setItem('nav-filter', newValue);
+          //Clear the filter query when filtertext is empty
+          if (!newValue) {
+            this.$router.replace({ path: this.$route.path, query: {} });
+          } else {
+            //else ,update the filter query param
+            this.$router.push({
+              path: this.$route.path,
+              query: { ...this.$route.query, filter: newValue },
+            });
+          }
+          this.debouncedUpdateQuery(newValue);
         }
       },
     },
     mounted() {
       if (window) {
-        const filterText = window.sessionStorage.getItem('nav-filter');
-        if (filterText) {
-          this.filterText = filterText;
+        const { filter } = this.$route.query;
+        // Set filterText from the query parameter if it exists
+        if (filter) {
+          this.filterText = filter;
         }
         this.$refs.links.scrollTop = window.sessionStorage.getItem('nav-scroll');
+        // Restoring filter state when a user navigates back
+        window.addEventListener('popstate', event => {
+          if (event.state && 'filterText' in event.state) {
+            this.filterText = event.state.filterText;
+          } else {
+            this.filterText = ''; // Reset if no filterText is in state
+          }
+        });
+        // Create debounced function for updating the query
+        this.debouncedUpdateQuery = debounce(newValue => {
+          if (newValue) {
+            this.$router.push({
+              path: this.$route.path,
+              query: { ...this.$route.query, filter: newValue },
+            });
+          } else {
+            this.$router.push({ path: this.$route.path, query: {} });
+          }
+        }, 2000); // 2-second debounce delay
```

Comment:
As we wont be using this debouncedUpdateQuery anymore, we can just remove these lines.
```suggestion
```